extensions/functions_string.yaml

```diff
@@ -403,3 +403,18 @@ scalar_functions:
           - value: "string"
           - value: i32
         return: "string"
+
+aggregate_functions:
+
+  -
+    name: string_agg
+    description: Concatenates a column of string values with a separator.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+            name: "input"
+            description: "Column of string values."
+          - value: "varchar<L2>"
+            name: "separator"
+            description: "Separator for concatenated strings"
+        return: "varchar<L1 + L2>"
```

Comment:
I'm not sure where L1 + L2 came from, but it seems rather arbitrary; "the resulting max length is that of the max length of one value and one separator"? `varchar<L1>` would make more sense to me. Also, why define this for `varchar`s and not for `string`s (or for both)?

Reply:
I added string